### DISCUSSION TITLE
IFC with function definitions

### DISCUSSION THREAD

**Discussion by madhurajith** [microsoft/ifc-spec#40] [Q&A] [answered]
Is it possible to generate an IFC with function definitions in it from MSVC?

**Comment by gdr-at-ms** [comment, marked as answer]
It is possible to do that for a few scenarios (like exported constexpr functions, or constexpr functions in header units).  There is active ask to the MSVC team to provide a compiler switch for generating IFC for any function definitions.  See this [community ask](https://developercommunity.visualstudio.com/t/add-support-to-clexe-to-create-an-ifc-file-directl/1293280?from=email&viewtype=all).  Please, have a look at it and upvote as you see fit.

Longer term, I would like to see all C++ compiler provide facilities like that.
